packages/app/hooks/use-persist-form.ts

```diff
@@ -38,87 +37,133 @@ export const usePersistForm = (
 ) => {
   const watchedValues = watch();
   const fileStorage = useMemo(() => new FileStorage(name), [name]);
-
+  let persistDebounceTimeout = useRef<any>(null);
```

Comment:
also added debounce when saving to localStorage. Before it used to store on each keystroke.